src/service/lib/get-mock-data.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const fs = require(`fs`).promises;
+
+const {
+  MOCK_FILE_PATH,
+} = require(`../constants`);
+
+let data = null;
+
+const getMockData = async () => {
+  if (data !== null) {
+    return Promise.resolve(data);
+  }
+
+  try {
+    const fileContent = await fs.readFile(MOCK_FILE_PATH);
+    data = JSON.parse(fileContent);
+  } catch (err) {
+    return Promise.reject(err);
+  }
+
+  return Promise.resolve(data);
```

Comment:
```suggestion
  if (data !== null) {
    return data;
  }

    const fileContent = await fs.readFile(MOCK_FILE_PATH);
    data = JSON.parse(fileContent);

  return data;
```
функция итак async, лишние обертки никчесму

Reply:
Поясню. Код брал из https://up.htmlacademy.ru/nodejs/3/book/4.13 раздел Разработка API. 
Единственное, что мы вызываем её асинхронно в итоге `const mockData = await getMockData();`, а значит она должна всегда возвращать Promise. Вроде так...

Reply:
Асинхронная функция всегда возвращает промисс

Reply:
Да, конечно. Я про блоки выше и ниже readFile
```
if (data !== null) {
    return data;
  }
```
и 
```
return data;
```

Хотя, попробовал сейчас, нормально отрабатывает. Спасибо!


Но try/catch я бы оставил, для отлова падения JSON.parse(), что, к сожалению, не является редкостью.

Reply:
Так а какой смысл от этого трайкеча? ты кидаешь эту ошибку дальше. Ты же её не обрабатываешь. Тоесть от него сейчас пользы 0. Такое же поведение будет без этого.

Reply:
Ну тоже верно. Спасибо!
Пример в учебники немного смущает

Reply:
Пример в учебнике это эталон (просто пример). Там не обязательно то, как должно быть 100%. Нужно тоже думать, так сказать 🙂